src/main/scala/io/getclump/ClumpContext.scala

```diff
@@ -7,31 +7,47 @@ private[getclump] final class ClumpContext {
   private[this] val fetchers =
     new HashMap[ClumpSource[_, _], ClumpFetcher[_, _]]()
 
-  def flush(clumps: List[Clump[_]]): Future[Unit] =
+  def flush(clumps: List[Clump[_]]): Future[Unit] = {
+    // 1. Get a list of all visible clumps grouped by level of composition
+    val clumpsByLevel = getClumpsByLevel(clumps)
+
+    // 2. Flush the fetches from all the visible clumps
+    flushFetches(clumpsByLevel.flatten).flatMap { _ =>
+      // 3. Walk through the downstream clumps as well, starting at the deepest level
+      flushDownstreamByLevel(clumpsByLevel.reverse)
+    }
+  }
+
+  // Unfold all visible (ie. upstream) clumps
+  private[this] def getClumpsByLevel(clumps: List[Clump[_]]): List[List[Clump[_]]] = {
```

Comment:
not a big deal at all but just wondering if for clarity it could be `upstreamByLevel` or `upstreamByLowestToHighestLevel` or something